### PR TITLE
Allow object for key signing

### PR DESCRIPTION
**Pull request by alexandrebodin**
This allows using passphrase protection for rsa keys

### CODE REVIEW

**Comment by omsmith** [comment]
You've a typo in the verifier typeof check

(On my phone, commenting on the line directly is hard)

**Comment by alexandrebodin** [comment]
Thanks! 

I have a problem with the version of the crypto api... in node v10.39 the sign method expects a buffer or a string... but in node v0.11.8+ objects are allowed...

**Comment by omsmith** [comment]
Mind adding a `process.version` check in the tests or somesuch, so that we can still run the tests in 0.10?

Would either not run the password-protected PEM tests, or would assert that they throw 'Not a buffer'

File: README.md [hunk 1]
```diff
@@ -74,6 +74,16 @@ the meantime, you can use
 [brianloveswords/base64url](https://github.com/brianloveswords/base64url)
 to decode the signature.
 
+As of nodejs `v0.11.8`, SPKAC support was introduce. If your node
+version satisfies, then you can pass an object 
+
+```javascript
```
Comment (omsmith) [inline README.md]: (nit) Mind just making this inline with the sentence? It takes up quite a bit of visual space without providing a whole ton of value

File: README.md [hunk 2]
```diff
@@ -74,6 +74,16 @@ the meantime, you can use
 [brianloveswords/base64url](https://github.com/brianloveswords/base64url)
 to decode the signature.
 
+As of nodejs `v0.11.8`, SPKAC support was introduce. If your node
```
Comment (omsmith) [inline README.md]: (nit) I think the verison number would be better represented as just emphasized, as it's not code, *v_0.11.8*

File: index.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ const util = require('util');
 
 const MSG_INVALID_ALGORITHM = '"%s" is not a valid algorithm.\n  Supported algorithms are:\n  "HS256", "HS384", "HS512", "RS256", "RS384", "RS512" and "none".'
 const MSG_INVALID_SECRET = 'secret must be a string or buffer';
-const MSG_INVALID_KEY = 'key must be a string or buffer';
+const MSG_INVALID_KEY = 'key must be a string, a buffer or an object';
```
Comment (omsmith) [inline index.js]: This message is being reused for both `sign` and `verify`, and as such is not incorrect for `verify`

**Comment by alexandrebodin** [comment]
On it

**Comment by omsmith** [comment]
@brianloveswords This LGTM. Would like to publish current master as a patch version before merging this as a minor.

**Comment by omsmith** [comment]
@brianloveswords FYI, I don't have npm access so I can't action the above

**Comment by brianloveswords** [comment]
@omsmith added you as co-owner of `jws` and `jwa` !

**Pull request merged**